frontend/src/components/createProjectForm.tsx

```diff
@@ -0,0 +1,366 @@
+import useTranslation from "next-translate/useTranslation";
+import { Badge, Col, FormSelect, Row } from "react-bootstrap";
+import styles from "../styles/createProjectForm.module.css";
+import { Field, Form, Formik, useFormik } from "formik";
+import useSWR from "swr";
+import apiPaths from "../properties/apiPaths";
+import { getAllUsersFromPage } from "../api/calls/userCalls";
+import { capitalize } from "../utility/stringUtil";
+import {
+    FormSubmitValues,
+    ProjectCreationProps,
+    ProjectCreationValues,
+} from "../handlers/createProjectSubmitHandler";
+import { IUser } from "../api/entities/UserEntity";
+import { getAllSkillTypesFromPage } from "../api/calls/skillTypeCalls";
+import { getSkillColorMap, ISkillType } from "../api/entities/SkillTypeEntity";
+import { ChangeEvent, useState } from "react";
+import Image from "next/image";
+import { useSwrWithEdition } from "../hooks/utilHooks";
+import useEdition from "../hooks/useGlobalEdition";
+import { getEntityFromFullUrl } from "../api/calls/baseCalls";
+
+export const CreateProjectForm = (props: ProjectCreationProps) => {
+    let userResponse = useSwrWithEdition(apiPaths.users, getAllUsersFromPage);
+
+    let users: IUser[] = userResponse.data || [];
+    let userError: Error = userResponse.error;
+
+    let skillTypeResponse = useSwrWithEdition(apiPaths.skillTypes, getAllSkillTypesFromPage);
+
+    let skillTypes: ISkillType[] = skillTypeResponse.data || [];
+    let skillTypeError: Error = skillTypeResponse.error;
+
+    const { t } = useTranslation("common");
+    const [edition] = useEdition();
+
+    const [goals, setGoals] = useState<string[]>([]);
+    const [goalInput, setGoalInput] = useState<string>("");
+    const [coaches, setCoaches] = useState<string[]>([]);
+    const [selectedCoach, setSelectedCoach] = useState<string>("");
+    const [skills, setSkills] = useState<string[]>([]);
+    const [selectedSkill, setSelectedSkill] = useState<string>("");
+    const [skillInfos, setSkillInfos] = useState<string[]>([]);
+    const [skillInfo, setSkillInfo] = useState<string>("");
```

Comment:
Isn't formik already changeling this state?

Reply:
There were complications with the creation of this component and I would rather not rework a significant part of it, but I will take a look if I find the time